### DISCUSSION TITLE
[Kavita] Improved Parsing for Western Comics (Library=Comics)

### DISCUSSION THREAD

**Discussion by DieselTech** [Kareadita/Kavita#2542] [Ideas]
### Idea Description

Posted by David Stevenson
· 5 months ago
· 24 Upvotes
https://feats.kavitareader.com/posts/396/improved-parsing-for-western-comics-library-comics

Description

Summary:
Refinement in the parsing system for western comics, especially for #0 and #00 issues.

Current Behavior:
At present, Kavita places the #0 and #00 issues under the 'Specials' tab.

Expected Behavior:
Issues labeled as #0 and #00 should be categorized under the 'Issues' tab. The 'Specials' tab should exclusively host files that have the metadata &lt;format&gt; tag populated with designations such as “Annual”, “TPB”, “Special”, etc.

Rationale:

Authenticity: By convention in western comics, #0 or #00 issues often serve as prequels or introductions to the main series. Thus, logically, they belong to the main issue lineup.
User Experience: Users familiar with western comic nomenclature might find the current system counter-intuitive or confusing. Having a clear distinction between main issues and specials would improve navigability and user satisfaction.

Implementation Suggestion:

Update the parsing algorithm to recognize #0 and #00 as part of the 'Issues' category.
Retain or refine the existing mechanism for populating the 'Specials' tab, using the metadata &lt;format&gt; tag as the primary criterion.

Impact:
This change would enhance the user experience for those who frequently read western comics on Kavita. It would ensure that the comics are organized in a way that aligns with the standard conventions of the genre.

Thank you for considering this feature request.

### Duration of Using Kavita

42

### Idea Category

Feature Enhancement

### Agreement

- [X] I agree that this is solely for submitting ideas, and I will search for existing ideas before posting.

**Comment by XyberDAWG** [comment]
One addition for Western Comics, Imprints are a large part of Western Comics, Parsing the <Imprint> tag from the comicinfo file as either a separate "Imprint" entry on the People page, or as a secondary Publisher would be incredibly Useful.

**Reply by majora2007** [reply]
Could you help me understand what this means and what features could be done with this tag?

**Reply by Auverus00** [reply]
Big comic publishers often have Imprints, which are more or less smaller publishers within that company. The big example is probably DC Comics' Vertigo Imprint, which gave us the classic Sandman and Swamp Thing comics, but also things like Wildstorm, Milestone, etc.

**Reply by therobbiedavis** [reply]
The only things I can really think that can be done with this information is for searching/filtering, and some people may want an imprint badge or some indication in the ui that the series is from an imprint, especially if they keep all DC (including imprints) in one Kavita library.

**Comment by Tomomatoes** [comment]
There are some additions that I would like to see with regards to western comics:
- Support for negative issue numbers. Some of the Spider-Man series have a `-1` issue, they appear in the volumes tab at the beginning like they should, but the issue does some seem to be present in the issues tab.
- Annuals aren't getting detected for my reading lists. In my case The Amazing Spider-man Annual. I wonder if this could be fixed with the format tag.
- Support for non integer issue numbers. There are issues within The Amazing Spider-Man V2018 such as issue `18HU` that don't appear in the issues tab. Said issues appear within the volume tab but at the beginning of the list rather than between issues 18 and 19.
- Not sure whether this is in the scope of the project, but would also like to see different volumes of series seperated or at least different screens within the series for metadata. 
Sticking with the Spider-man examples, there are 6 volumes, the summary meta data refers to only V1 (1963) and all the other meta data is merged from the remaining 5 volumes. Unlike in Manga, these volumes aren't necessarily related to each other.

**Reply by majora2007** [reply]
Hi @Tomomatoes I am really currious how this works: "Support for non integer issue numbers. There are issues within The Amazing Spider-Man V2018 such as issue 18HU that don't appear in the issues tab. Said issues appear within the volume tab but at the beginning of the list rather than between issues 18 and 19."

To my knowledge, there are other non-number based issue numbers like an omega symbol, a 1/2 symbol, infinity symbol, etc. To my knowledge there is no way to order these correctly as Omega might be between different issues. 

How would you suggest this be tackled? I was thinking of introducing a new tag to ComicInfo for IssueOrder which can be filled out. So issue 18HU would be 18.5. If the tag isn't there, ordering would use Issue number by default. This could theoretically allow other systems besides Kavita from implementing proper ordering and not have to rely on reading lists.

**Reply by XyberDAWG** [reply]
This would actually work well. Omega is usually at the end of a volume, Alpha is usually at the beginning of a volume (Aka 0) they are usually used as a Prologue and Epilogue of an Event.

**Reply by Tomomatoes** [reply]
"I was thinking of introducing a new tag to ComicInfo for IssueOrder which can be filled out. So issue 18HU would be 18.5. If the tag isn't there, ordering would use Issue number by default. This could theoretically allow other systems besides Kavita from implementing proper ordering and not have to rely on reading lists."
I think this is a good solution, I didn't have any ideas myself, trying to implement work arounds for the parser could be messy and error prone, so this seems like a good alternative. From my library I don't have many comics that contain these non integer characters so this wouldn't be much work to add manually.

I do appreciate that the numbering for comic issues can get ridiculous.

I am willing to assist in testing when the time comes, I have quite a large library of comics.

**Reply by XyberDAWG** [reply]
I too am willing to test this feature.

**Comment by majora2007** [comment]
I have an issue on my plan to tackle this up here:
https://github.com/Kareadita/Kavita/issues/2714

Please comment here in the discussion if there are any concerns with this.

**Comment by majora2007** [comment]
If anyone here would like to help me test the canary that has this implemented, please do. Please see:
https://github.com/Kareadita/Kavita/pull/2772

All details for testing are in the PR. It is crucial to provide feedback earlier. Once this hits nightly, it is unlikely to change in the core.

**Comment by majora2007** [comment]
This has been completed as of https://github.com/Kareadita/Kavita/pull/2796

It will be available in the v0.8.0 release.
